hypervisor/dm/vioapic.c

```diff
@@ -326,6 +328,13 @@ static void vioapic_indirect_write(struct acrn_vioapic *vioapic, uint32_t addr,
 			dev_dbg(ACRN_DBG_IOAPIC, "ioapic pin%hhu: redir table entry %#lx",
 				pin, vioapic->rtbl[pin].full);
 
+			/* remap for ptdev */
+			if ((new.bits.intr_mask == IOAPIC_RTE_MASK_CLR) || (last.bits.intr_mask  == IOAPIC_RTE_MASK_CLR)) {
+				/* VM enable intr */
+				/* NOTE: only support max 256 pin */
+				(void)ptirq_intx_pin_remap(vioapic->vm, pin, PTDEV_VPIN_IOAPIC);
+			}
+
```

Comment:
I assume this is the actual fix, as vioapic_generate_intr may trigger interrupt without correct pin state config. Could you double check if only this code change fixed your problem?